server/src/main/java/org/elasticsearch/index/engine/LazySoftDeletesDirectoryReaderWrapper.java

```diff
@@ -260,4 +276,26 @@ public boolean initialized() {
             return materializedBits != null;
         }
     }
+
+    private static CacheKey getCacheKey(DirectoryReader reader, String field) throws IOException {
+        SoftDeletesDirectoryReaderWrapper wrapper = new SoftDeletesDirectoryReaderWrapper(reader, field);
```

Comment:
Hm, I realized there's an issue if reader.getReaderCacheHelper() is null. I can easily work around the null pointer here, but I need to somehow get a key inside the leaf and codec readers.